tests/pytests/test_acl.py

```diff
@@ -204,3 +204,72 @@ def test_internal_commands(env):
     # Authenticate as `default`, and run the internal debug command
     env.expect('AUTH', 'default', 'nopass').true()
     env.expect(debug_cmd(), 'DUMP_TERMS', 'idx').equal([])
+
+@skip(redis_less_than="7.9.0")
+def test_acl_key_permissions_validation(env):
+    """Tests that the key permission validation works properly"""
+
+    # Create an ACL user with partial key-space permissions
+    env.expect('ACL', 'SETUSER', 'test', 'on', '>123', '~h:*', '&*', '+@all').ok()
+
+    # Create an index on the key the user does not have permissions to access
+    # All prefixes will do (default)
+    idx_name = 'idx'
+    env.expect('FT.CREATE', idx_name, 'SCHEMA', 'n', 'NUMERIC')
```

Comment:
Consider testing another index that has prefix `h:a` that could be used
And another index that has prefix `k:` that cannot be used